src/malli/core.cljc

```diff
@@ -122,6 +119,8 @@
 
 (defn -update [m k f] (assoc m k (f (get m k))))
 
+(defn -memoize [f] (let [value (atom nil)] (fn [] (or @value) (reset! value (f)))))
```

Comment:
Extracted from `-ref-schema`, but why not just use `delay`?

Reply:
no idea. maybe it should.